crates/tools/yml/src/main.rs

```diff
@@ -137,3 +138,65 @@ jobs:
 
     std::fs::write(".github/workflows/clippy.yml", yml.as_bytes()).unwrap();
 }
+
+// Ideally this would just be another matrix dimension in test_yml but Cargo limitations are blocking this for the time being.
```

Comment:
I don't understand why the Cargo limitation blocks this

Reply:
As noted above, if you're cross compiling you need to use `--target` and then the default placement of build artifacts is disrupted and the component (DLL client and server) tests would fail. More info in specific comment:

https://github.com/rust-lang/cargo/issues/9661#issuecomment-2059686735

Reply:
There's hope in "artifact dependencies" but this feature is not stable and doesn't yet work very well. See this comment:

https://github.com/rust-lang/cargo/issues/9661#issuecomment-2059974262